File: knowledge/scaffolds/web-backend/go-http/handlers/handlers.go
package handlers

import (
	"encoding/json"
	"net/http"
)

type MessageResponse struct {
	Message string `json:"message"`
}

type HealthResponse struct {
	Status string `json:"status"`
}

func Hello(w http.ResponseWriter, r *http.Request) {
	writeJSON(w, http.StatusOK, MessageResponse{Message: "Hello, World!"})
}

func HelloName(w http.ResponseWriter, r *http.Request) {
	name := r.PathValue("name")
	writeJSON(w, http.StatusOK, MessageResponse{Message: "Hello, " + name + "!"})
}

func Health(w http.ResponseWriter, r *http.Request) {
	writeJSON(w, http.StatusOK, HealthResponse{Status: "ok"})
}

func writeJSON(w http.ResponseWriter, status int, data any) {
	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(status)
	json.NewEncoder(w).Encode(data)
}

File: knowledge/scaffolds/web-backend/rust-actix/src/handlers.rs
use actix_web::{web, HttpResponse};
use crate::models::{MessageResponse, HealthResponse};

pub async fn hello() -> HttpResponse {
    let response = MessageResponse {
        message: "Hello, World!".to_string(),
    };
    HttpResponse::Ok().json(response)
}

pub async fn hello_name(path: web::Path<String>) -> HttpResponse {
    let name = path.into_inner();
    let response = MessageResponse {
        message: format!("Hello, {}!", name),
    };
    HttpResponse::Ok().json(response)
}

pub async fn health() -> HttpResponse {
    let response = HealthResponse {
        status: "ok".to_string(),
    };
    HttpResponse::Ok().json(response)
}
